spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -268,17 +650,349 @@ public PublisherBuilder<IncomingMessage, OutgoingMessage> processMessages() {
 
 ==== Message acknowledgement
 
+Acknowledgement is an important part of message processing.
 Messages are either acknowledged explicitly, or implicitly by the implementation.
 
-===== Explicit acknowledgement
+Acknowledgement is controlled by the `org.eclipse.microprofile.reactive.messaging.Acknowledgment` annotation. 
+The annotation allows configuring the acknowledgement strategy among:
+
+* `NONE` - no acknowledgment is performed
+* `MANUAL` - the user is responsible for the acknowledgement, so the Reactive Messaging implementation does not apply implicit acknowledgement
+* `PRE_PROCESSING` - the Reactive Messaging implementation acknowledges the message before the annotated method or processing is executed
+* `POST_PROCESSING` - the Reactive Messaging implementation acknowledges the message once:
+
+  1. the method or processing completes if the method does not emit data
+  2. when the emitted data is acknowledged
+
+Each method signatures has different defaults and accept a different set of strategy.
```

Comment:
Each method signature type has different defaults and can implement different acknowledgement policies.